### PR TITLE
Add support for OTLP HTTP API.

### PR DESCRIPTION
**Pull request by fmassot**
- [x] add tests

I tested the OTLP HTTP API with tracegen for traces and with an OTEL collector for logs (compression must be set to none).

### CODE REVIEW

**Comment by etolbakov** [comment]
All LGTM

File: quickwit/quickwit-serve/src/otlp_api/rest_handler.rs [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+    otlp_traces_service: Option<OtlpGrpcTracesService>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    oltp_ingest_logs_handler(otlp_logs_service).or(oltp_ingest_traces_handler(otlp_traces_service))
+}
+
+pub(crate) fn oltp_ingest_logs_handler(
+    otlp_log_service: Option<OtlpGrpcLogsService>,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    require(otlp_log_service)
+        .and(warp::path!(String / "ingest" / "otlp" / "v1" / "logs"))
```
Comment (guilload) [inline quickwit/quickwit-serve/src/otlp_api/rest_handler.rs:54]: Feature request:
Support the root endpoint `host:port/otlp` to ingest directly in the default logs/traces indexes.

File: quickwit/quickwit-serve/src/otlp_api/rest_handler.rs [hunk 2]
```diff
@@ -0,0 +1,255 @@
...
+        .and(warp::header::exact_ignore_case(
```
Comment (guilload) [inline quickwit/quickwit-serve/src/otlp_api/rest_handler.rs:55]: Feature request:
Support JSON over HTTP in addition to Proto.

**Review by guilload — APPROVED** [review]

File: quickwit/quickwit-serve/src/otlp_api/rest_handler.rs [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+use warp::{Filter, Rejection};
+
+use crate::json_api_response::make_json_api_response;
+use crate::{require, with_arg, BodyFormat};
+
+#[derive(utoipa::OpenApi)]
+#[openapi(paths())]
+pub(crate) struct OtlpApi;
+
+/// Setup OpenTelemetry API handlers.
```
Comment (etolbakov) [inline quickwit/quickwit-serve/src/otlp_api/rest_handler.rs:42]: Should we make an update in `openapi.rs`?

Comment (fmassot) [reply quickwit/quickwit-serve/src/otlp_api/rest_handler.rs:42]: Yes I did not implement it here. Do you want to add it in a follow up PR?

Comment (etolbakov) [reply quickwit/quickwit-serve/src/otlp_api/rest_handler.rs:42]: yeah, happy to do that later today

**Review by etolbakov — APPROVED** [review]

**Pull request merged**